### PR TITLE
ParticleUpdater Tests

### PR DESCRIPTION
**Pull request by MaxBorsch**
Adds a few tests for `ParticleUpdater.java` from the particle system I worked on last year. There's still many more tests to write but I wanted to avoid adding tests for things that may be overhauled in the foreseeable future.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine-tests/src/test/java/org/terasology/particles/updating/ParticleUpdaterImplTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+/**
+ * Unit test for {@link ParticleUpdaterImpl}.
+ */
+public class ParticleUpdaterImplTest {
+
+	private ParticleUpdater particleUpdater;
+	private BiMap<Class<Component>, GeneratorFunction> registeredGeneratorFunctions;
+	private BiMap<Class<Component>, AffectorFunction> registeredAffectorFunctions;
+
+	@Before
```
Comment (vampcat) [inline engine-tests/src/test/java/org/terasology/particles/updating/ParticleUpdaterImplTest.java]: Tabs? I'm disappointed.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by vampcat** [comment]
So, I was going through the code, one thing stuck out: In ParticlePoolTest.java, IntelliJ complains about value never being used on line 188. Can you have a look?

**Comment by MaxBorsch** [comment]
@vampcat Fixed it. It was a left over line from the previous block being copy/pasted. There is a comment at the end of the chain of assertions that says `// there are no living particles to compare here` and it replaces the line that would compare the two pools in the previous block that was copy/pasted.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by Cervator** [comment]
Ran tests, removed stray file from under configs, merged. Thanks :-)

**Pull request merged**